File: aula-extra-est-main/BackendAulaExtra/src/Modules/Student/DTOs/PendingProfessorEvaluationDto.cs
using System;

namespace ConfidantPostgreSQL.Modules.Student.DTOs
{
    public class PendingProfessorEvaluationDto
    {
        public Guid ProfessorId { get; set; }
        public string ProfessorName { get; set; } = string.Empty;
        public DateTime? LastLessonStart { get; set; }
        public DateTime? LastLessonEnd { get; set; }
    }
}

File: aula-extra-est-main/FrontendAulaExtra/AulaExtra/lib/core/data/student_professor_evaluations/dtos/pending_professor_evaluation_dto.dart
class PendingProfessorEvaluationDto {
  PendingProfessorEvaluationDto({
    required this.professorId,
    required this.professorName,
    required this.lastLessonStart,
    required this.lastLessonEnd,
  });

  final String professorId;
  final String professorName;
  final DateTime? lastLessonStart;
  final DateTime? lastLessonEnd;

  factory PendingProfessorEvaluationDto.fromJson(Map<String, dynamic> json) {
    return PendingProfessorEvaluationDto(
      professorId: json['professorId']?.toString() ?? '',
      professorName: json['professorName']?.toString() ?? '',
      lastLessonStart: json['lastLessonStart'] == null ? null : DateTime.tryParse(json['lastLessonStart'].toString()),
      lastLessonEnd: json['lastLessonEnd'] == null ? null : DateTime.tryParse(json['lastLessonEnd'].toString()),
    );
  }
}
